src/x86_64/service.c

```diff
@@ -74,10 +75,19 @@ void process_bhqueue()
 {
     /* XXX - we're on bh frame & stack; re-enable ints here */
     thunk t;
-    while((t = dequeue(bhqueue))) {
```

Comment:
Does it make more sense to process the deferqueue first, so that if it generates work on the bhqueue we'll be able to process it? This would of course depend on how we decide to use the two queues, so it's not clear that such a scenario will happen, but it might be important for making sure that any time-sensitive work that is spawned off of the deferqueue is processed quickly

Reply:
So the idea was that the bhqueue is used for high priority events triggered by interrupt handlers and other exceptions. The deferqueue is for when a condition requires blocking (deferred execution - e.g. deferring the sending of output data for a connection when the TCP send window is full). It's more likely that an event triggered by the bhqueue will queue up work on the deferqueue rather than the other way around.